sssd_test_framework/utils/socket.py

```diff
@@ -0,0 +1,245 @@
+"""Manage socket-activated responders in SSSD for testing purposes."""
+
+from __future__ import annotations
+
+import re
+from typing import Final, Optional
+
+from pytest_mh import MultihostHost, MultihostUtility
+from pytest_mh.conn import ProcessResult
+
+__all__ = [
+    "SSSDSocketResponder",
+    "SSSDSocketUtils",
+    "ResponderType",
+]
+
+
+class ResponderType(str):
+    """
+    Well-known SSSD responders that support socket activation.
+    """
+
+    NSS: Final = "nss"
+    """Name Service Switch responder"""
+
+    PAM: Final = "pam"
+    """Pluggable Authentication Module responder"""
+
+    SUDO: Final = "sudo"
+    """Sudo responder"""
+
+    SSH: Final = "ssh"
+    """SSH responder"""
+
+    PAC: Final = "pac"
+    """Privilege Attribute Certificate responder"""
+
+    AUTOFS: Final = "autofs"
+    """Automount responder"""
+
+
+class SSSDSocketResponder:
+    """
+    Represents an SSSD socket-activated responder (e.g., nss, pam).
+    Provides methods to manage its socket and service units.
+    """
+
+    def __init__(
+        self,
+        host: MultihostHost,
+        responder: str,
+    ) -> None:
+        """
+        Initialize a socket-activated responder object.
+
+        :param host: The host where SSSD services run.
+        :type host: MultihostHost
+        :param responder: The responder type (e.g., 'nss', 'pam').
+        :type responder: str
+        """
+        if responder not in [
+            ResponderType.NSS,
+            ResponderType.PAM,
+            ResponderType.SUDO,
+            ResponderType.SSH,
+            ResponderType.PAC,
+            ResponderType.AUTOFS,
+        ]:
+            raise ValueError(f"Unknown responder: {responder}")
+
+        self.host = host
+        self.responder = responder
+        self.socket_unit = f"sssd-{responder}.socket"
+        self.service_unit = f"sssd-{responder}.service"
+
+    def enable_socket(self) -> None:
+        """
+        Enable the responder's socket unit.
+        """
+        result = self.host.conn.exec(["systemctl", "enable", self.socket_unit], raise_on_error=False)
```

Comment:
All these calls are already implemented in [SystemdServices](https://pytest-mh.readthedocs.io/en/latest/api/pytest_mh.utils.services.html#pytest_mh.utils.services.SystemdServices) with error handling, logging and restoration. You should pass svc to this class and use these calls.

Reply:
All of the methods are not available via svc. So not implementing